config/initializers/devise.rb

```diff
@@ -29,7 +29,7 @@
   # session. If you need permissions, you should implement that in a before filter.
   # You can also supply a hash where the value is a boolean determining whether
   # or not authentication should be aborted when the value is not present.
-  config.authentication_keys = [:email]
+  config.authentication_keys = [:email, :firstname, :lastname]
```

Comment:
Are you not supposed to add the keys to the `config.strip_whitespace_keys`? :D

Reply:
Yes, obviously. What did I do?